### ISSUE TITLE
ReplacementContext::printInst does not display single consts

### ISSUE THREAD

**Issue by zhengyangl** [google/souper#339]
This is rather a minor issue, but might cause confusion in future development.

A little bit context: ReplacementContext::printInst() is designed in a recursive way: Each call to the function returns the "name" of the instruction and dump the input inst on the screen. If the instruction has operands, then call the function it self with each operand input, dump the operands, then with the name of the operands, it composites the single line output of the instruction and dumps.

 To facilitate the compact display of Souper instructions, the constants are designed to dump nothing but just returns the value of the constant with its type. For example: 
```
%5 = mul %4, 0: i32 
```

Consider the code
```
Inst *Ante = IC.getConst(APInt(1, true));
ReplacementContext RC;
RC.printInst(Ante, llvm::errs(), false);
```
Since we're dumping a single constant, there will be nothing on the screen. This caused come confusion to me. 

Some grammar like this might be good:
```
%5 = constant 0:i32
%6 = mul %4, %5(0:i32)
```

**Comment by regehr** [comment]
yeah I've run into this too.
I don't think we should change anything by default, since this will reduce the compactness of souper IR and increase the size of the redis DB, but if you want to print the alternate version when a specific flag is provided, that seems reasonable.
also you would need to decide if you want to parse the alternate format, or if it is print-only

**Comment by zhengyangl** [comment]
@regehr Yeah sure, I agree with you for not making any change on grammar. I think something like dump() within Souper::Inst itself may be a good choice for debug use.
